packages/ra-ui-materialui/src/form/SimpleFormIterator.js

```diff
@@ -99,7 +99,7 @@ export class SimpleFormIterator extends Component {
     addField = () => {
         const { fields } = this.props;
         this.ids.push(this.nextId++);
-        fields.push({});
+        fields.push();
```

Comment:
Isn't this equivalent to not calling push at all?

Reply:
No, without this it won't add anything

Reply:
I think calling `array.push` without any argument does not modify the array and is equivalent to not calling the function at all. At least this is what I see when trying this in a console

Reply:
@djhi You're right 
![image](https://user-images.githubusercontent.com/1819833/58029158-506afb80-7b1c-11e9-86c9-d190490016be.png)

Reply:
It would be right if `fields` were just a standard mutating array, however it's not.
See here https://redux-form.com/8.2.0/docs/api/fieldarray.md/#-code-fields-push-value-any-function-code-